src/cockpit/389-console/src/lib/security/certificateManagement.jsx

```diff
@@ -165,6 +241,105 @@ export class CertificateManagement extends React.Component {
                 });
     }
 
+    addCSR () {
+        if (this.state.csrName == "") {
+            this.props.addNotification(
+                "warning",
+                `Missing CSR Name`
+            );
+            return;
+        } else if (this.state.csrSubjectCommonName == "") {
+            this.props.addNotification(
+                "warning",
+                `Missing CSR Subject Common Name`
+            );
+            return;
+        }
+
+        this.setState({
+            modalSpinning: true,
+            loading: true,
+        });
+        const cmd = [
+            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+            "security", "csr", "req", "--name=" + this.state.csrName, "--subject=" + this.state.csrSubject
+        ];
+        log_cmd("addCSR", "Creating CSR", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done(() => {
+                    this.reloadCSRs();
+                    this.setState({
+                        showAddCSRModal: false,
+                        csrSubject: '',
+                        csrName: '',
+                        modalSpinning: false
+                    });
+                    this.props.addNotification(
+                        "success",
+                        `Successfully created CSR`
+                    );
+                })
+                .fail(err => {
+                    const errMsg = JSON.parse(err);
+                    let msg = errMsg.desc;
+                    if ('info' in errMsg) {
+                        msg = errMsg.desc + " - " + errMsg.info;
+                    }
+                    this.setState({
+                        modalSpinning: false,
+                        loading: false,
+                    });
+                    this.props.addNotification(
+                        "error",
+                        `Error creating CSR - ${msg}`
+                    );
+                });
+    }
+
+    showCSR (name) {
+        if (name == "") {
+            this.props.addNotification(
+                "warning",
+                `Missing CSR Name`
+            );
+            return;
+        }
+
+        this.setState({
+            modalSpinning: true,
+            loading: false,
+        });
+
+        const cmd = [
+            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+            "security", "csr", "get", name
+        ];
+
+        log_cmd("showCSR", "Displaying CSR", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message"})
+                .done(content => {
+                    this.setState({
+                        csrContent: content,
+                        showViewCSRModal: true,
+                        modalSpinning: true,
```

Comment:
modalSpinning needs to be set to false here.  After viewing a CSR the modal is still set as spinning.  This is why the create csr modal is spinning when I open it

Reply:
same with "loading", I don't need to set it.